includes/notes/class-wc-payments-notes-loan-approved.php

```diff
@@ -0,0 +1,165 @@
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class WC_Payments_Notes_Additional_Payment_Methods
+ */
+class WC_Payments_Notes_Loan_Approved {
+	use NoteTraits;
+
+	/**
+	 * Prefix of the note for use in the database.
+	 */
+	const NOTE_NAME = 'wc-payments-notes-loan-approved';
+
+	/**
+	 * Nonce action name
+	 */
+	const NOTE_ACTION = 'view-capital-page';
+
+	/**
+	 * Loan information to build the message.
+	 *
+	 * @var array
+	 */
+	private static $loan_info;
+
+	/**
+	 * Get the note.
+	 */
+	public static function get_note() {
+		$note_class = WC_Payment_Woo_Compat_Utils::get_note_class();
+		$note       = new $note_class();
+
+		$note->set_title( __( 'Your capital loan has been approved!', 'woocommerce-payments' ) );
+		$note->set_content(
+			sprintf(
+				// Translators: %1: total amount lent to the merchant formatted in the account currency.
+				__(
+					'Congratulations! Your capital loan has been approved and %1$s was deposited in to the bank account linked to WooCommerce Payments. You\'ll automatically repay the loan, plus a flat fee, through a fixed percentage of each WooCommerce Payments transaction.',
+					'woocommerce-payments'
+				),
+				WC_Payments_Explicit_Price_Formatter::get_explicit_price( wc_price( self::$loan_info['details']['advance_amount'] / 100 ) )
+			)
+		);
+
+		$note->set_content_data(
+			(object) [
+				'advance_amount'      => self::$loan_info['details']['advance_amount'],
+				'advance_paid_out_at' => self::$loan_info['details']['advance_paid_out_at'],
+			]
+		);
+		$note->set_type( $note_class::E_WC_ADMIN_NOTE_INFORMATIONAL );
+		$note->set_name( self::NOTE_NAME );
+		$note->set_source( 'woocommerce-payments' );
+		$note->add_action(
+			self::NOTE_NAME,
+			__( 'View loan details', 'woocommerce-payments' ),
+			admin_url( 'admin.php?page=wc-admin&path=/payments/loans' ),
+			$note_class::E_WC_ADMIN_NOTE_UNACTIONED,
+			true
+		);
+
+		return $note;
+	}
+
+	/**
+	 * Add the note if it passes predefined conditions.
+	 */
+	public static function possibly_add_note() {
+		// If we have the correct information, proceed. Otherwise, delete existing notes.
+		if ( ! self::validate_inputs() ) {
+			// We don't have the necessary info to create a note, do nothing.
+			return;
+		}
+
+		// Check if the current loan info matches with the received one. If it matches, don't add a new one.
+		if ( ! self::check_attached_loan_data_is_different() ) {
+			// Loan paid out dates are the same, do nothing.
+			return;
+		}
+
+		// Do the overridden work.
+		if ( ! self::can_be_added() ) {
+			return;
+		}
+
+		$new_note = self::get_note();
+		$new_note->save();
+	}
+
+	/**
+	 * Check if the stored loan info has all the values we need.
+	 *
+	 * @return bool
+	 */
+	private static function validate_inputs() {
+		// If the loan amount isn't set correctly, don't push the note, and delete the old one if exists.
+		if ( empty( self::$loan_info )
+			|| ! is_array( self::$loan_info )
```

Comment:
These 2 checks are redundant. `isset( self::$null_variable['something']['something_else] )` will just return `FALSE` without crashing or giving a warning.

Reply:
Removed in [cbb695c](https://github.com/Automattic/woocommerce-payments/pull/3780/commits/cbb695c836d638f93c87f874e664d4466caedea3).